examples/cpp/content_filter/CLIParser.hpp

```diff
@@ -0,0 +1,458 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <csignal>
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+
+#include <fastdds/dds/log/Log.hpp>
+
+#ifndef _FASTDDS_CONTENT_FILTER_CLI_PARSER_HPP_
+#define _FASTDDS_CONTENT_FILTER_CLI_PARSER_HPP_
+
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace content_filter {
+
+using dds::Log;
+
+class CLIParser
+{
+public:
+
+    CLIParser() = delete;
+
+    //! Entity kind enumeration
+    enum class EntityKind : uint8_t
+    {
+        PUBLISHER,
+        SUBSCRIBER,
+        UNDEFINED
+    };
+
+    //! Filter kind enumeration
+    enum class FIlterKind : uint8_t
+    {
+        DEFAULT,
+        CUSTOM,
+        NONE
+    };
+
+    //! Publisher configuration structure
+    struct publisher_config
+    {
+        uint16_t samples = 0;
+        uint16_t interval = 100;
+        bool filtering = true;
+    };
+
+    //! Subscriber application configuration structure
+    struct subscriber_config
+    {
+        CLIParser::FIlterKind filter_kind = CLIParser::FIlterKind::DEFAULT;
+        std::string upper_bound = "9";
+        std::string lower_bound = "5";
+    };
+
+    //! Configuration structure for the application
+    struct content_filter_config
+    {
+        CLIParser::EntityKind entity = CLIParser::EntityKind::UNDEFINED;
+        publisher_config pub_config;
+        subscriber_config sub_config;
+    };
+
+    /**
+     * @brief Print usage help message and exit with the given return code
+     *
+     * @param return_code return code to exit with
+     *
+     * @warning This method finishes the execution of the program with the input return code
+     */
+    static void print_help(
+            uint8_t return_code)
+    {
+        std::cout <<
```

Comment:
We should try leaving this help method more reduced so that it can be more legible, you can take the `hello_world` or `configuration` examples as a reference.

Reply:
I haven't realized that the uncrustify changed that part